internal/api/v1/services/user.go

```diff
@@ -0,0 +1,60 @@
+package services
+
+import (
+	"context"
+	"errors"
+
+	"github.com/celestiaorg/talis/internal/db/models"
+	"github.com/celestiaorg/talis/internal/db/repos"
+	"github.com/celestiaorg/talis/internal/types/infrastructure"
+)
+
+// User provides business logic for user operations
+type User struct {
+	repo *repos.UserRepository
+}
+
+// User service errors
+var (
+	ErrUserNotFound     = errors.New("user not found")
+	ErrUserCreateFailed = errors.New("failed to create user")
+)
+
+// NewUserService creates a new user service instance
+func NewUserService(repo *repos.UserRepository) *User {
+	return &User{
+		repo: repo,
+	}
+}
+
+// CreateUser creates a new user
+func (s User) CreateUser(ctx context.Context, userReq *infrastructure.CreateUserRequest) (uint, error) {
+	user := &models.User{
+		Username:     userReq.Username,
+		Email:        userReq.Email,
+		Role:         userReq.Role,
+		PublicSshKey: userReq.PublicSshKey,
+	}
```

Comment:
This request should be validated somewhere. I think the best place would be at the `repo/user.go` level. 

To start it should be verifying that the username name is not empty.